server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -379,6 +430,45 @@ public KeywordFieldMapper build(MapperBuilderContext context) {
                 this
             );
         }
+
+        private FieldType fieldType(final IndexSortConfig indexSortConfig, final IndexMode indexMode, final String fullFieldName) {
+            if (shouldUseDocValuesSparseIndex(indexSortConfig, indexMode, fullFieldName)) {
+                return new FieldType(Defaults.FIELD_TYPE_WITH_SKIP_DOC_VALUES);
+            }
+            return new FieldType(Defaults.FIELD_TYPE);
+        }
+
+        private boolean shouldUseDocValuesSparseIndex(
```

Comment:
I think we should also add a feature flag and check this here?

Reply:
AH right...I forgot about the feature flag.